apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/UnclaimedActivityRecordIntegrationTest.scala

```diff
@@ -0,0 +1,311 @@
+             new DsoRules_CreateUnallocatedUnclaimedActivityRecord(
+               aliceParty.toProtoPrimitive,
+               BigDecimal(amountToMint).bigDecimal,
+               "alice is doing great - vote",
+               Instant.now().plus(5, ChronoUnit.SECONDS),
+             )
+           )
+         )
+         eventuallySucceeds() {
+           sv1Backend.createVoteRequest(
+             sv1Party.toProtoPrimitive,
+             action,
+             "url",
+             "alice is doing great",
+             sv1Backend.getDsoInfo().dsoRules.payload.config.voteRequestTimeout,
+             None,
+           )
+         }
+       },
+     )("vote request and UnallocatedUnclaimedActivityRecord have been created",
+        _ => {
+         sv1Backend.listVoteRequests().loneElement
+          eventually(){
+           sv1Backend.participantClient.ledger_api_extensions.acs
+             .filterJava(UnallocatedUnclaimedActivityRecord.COMPANION)(dsoParty) should not be empty
+          }
+        }
+      )
+    }
+
+    setTriggersWithin(
+     triggersToPauseAtStart = allocateTriggers,
+     triggersToResumeAtStart = Seq.empty,
+    ) {
+      clue("UnallocatedUnclaimedActivityRecord gets archived"){
+        eventually(){
+         sv1Backend.participantClient.ledger_api_extensions.acs
+           .filterJava(UnallocatedUnclaimedActivityRecord.COMPANION)(dsoParty) shouldBe empty
+        }
+      }
+    }
+  }
+
+  "An UnclaimedActivityRecord gets expired" in { implicit env =>
+    val sv1UserId = sv1WalletClient.config.ledgerApiUser
+    val sv1Party = sv1Backend.getDsoInfo().svParty
+    val aliceUserName = aliceWalletClient.config.ledgerApiUser
+    val aliceParty = aliceValidatorBackend.onboardUser(aliceUserName)
+    val amountToMint = 15.0
+    val unclaimedRewardsToMint = Seq(15.0)
+
+    def allocateTriggers: Seq[Trigger] =
+      activeSvs.map(_.dsoDelegateBasedAutomation.trigger[AllocateUnallocatedUnclaimedActivityRecordTrigger])
+    def expiredUnallocatedTriggers: Seq[Trigger] =
+      activeSvs.map(_.dsoDelegateBasedAutomation.trigger[ExpiredUnallocatedUnclaimedActivityRecordTrigger])
+    def expiredUnclaimedTriggers: Seq[Trigger] =
+      activeSvs.map(_.dsoDelegateBasedAutomation.trigger[ExpiredUnclaimedActivityRecordTrigger])
+    // The trigger that merges amulets for alice
+    // Note: using `def`, as the trigger may be destroyed and recreated (when user is offboarded and onboarded)
+    def aliceMergeAmuletsTrigger: Trigger =
+      aliceValidatorBackend
+        .userWalletAutomation(aliceUserName)
+        .futureValue
+        .trigger[CollectRewardsAndMergeAmuletsTrigger]
+
+    actAndCheck(
+      "Mint some unclaimed rewards",
+      unclaimedRewardsToMint.foreach { amount =>
+        createUnclaimedReward(
+          sv1ValidatorBackend.participantClientWithAdminToken,
+          sv1UserId,
+          amount
+        )
+      },
+    )(
+      "Unclaimed rewards gets created",
```

Comment:
what does the wait here actually do? you're using a SubmitAndWait in createUnclaimedReward so it seems like that is already synchronous. If you do need mores ynchronization consider moving it directly into createUnclaimedReward